ionoscloud/resource_container_registry.go

```diff
@@ -0,0 +1,227 @@
+package ionoscloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	crService "github.com/ionos-cloud/terraform-provider-ionoscloud/v6/services/container-registry"
+	"github.com/ionos-cloud/terraform-provider-ionoscloud/v6/utils"
+	"log"
+	"regexp"
+	"time"
+)
+
+func resourceContainerRegistry() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceContainerRegistryCreate,
+		ReadContext:   resourceContainerRegistryRead,
+		UpdateContext: resourceContainerRegistryUpdate,
+		DeleteContext: resourceContainerRegistryDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceContainerRegistryImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"garbage_collection_schedule": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"time": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "UTC time of day e.g. 01:00:00 - as defined by partial-time - RFC3339",
+						},
+						"days": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Schema{
+								Type:             schema.TypeString,
+								ValidateDiagFunc: validation.ToDiagFunc(validation.IsDayOfTheWeek(true)),
+							},
+						},
+					},
+				},
+			},
+			"hostname": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"location": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+				ForceNew:         true,
+			},
+			"name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringMatch(regexp.MustCompile("^[a-z][-a-z0-9]{1,61}[a-z0-9]$"), "")),
```

Comment:
@cristiGuranIonos nice regex here, it matches exactly the standard that is presented in the documentation for the container registry name, I tested a little bit and the validation works properly.